### PR TITLE
Allow reference to original column name with alias

### PR DESCRIPTION
**Pull request by janetzki**
Close hyrise-mp/hyrise#36

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,6 +1,5 @@
-[![Build Status](https://hyrise-ci.epic-hpi.de/buildStatus/icon?job=Hyrise/hyrise/master)](https://hyrise-ci.epic-hpi.de/blue/organizations/jenkins/hyrise%2Fhyrise/activity/)
-[![Coverage Status](https://hyrise-coverage-badge.herokuapp.com/coverage_badge.svg)](https://hyrise-ci.epic-hpi.de/job/Hyrise/job/hyrise/job/master/lastStableBuild/Llvm-cov_5fReport/)
-[![CodeFactor](https://www.codefactor.io/repository/github/hyrise/hyrise/badge)](https://www.codefactor.io/repository/github/hyrise/hyrise)
+[![Build Status](https://hyrise-ci.epic-hpi.de/buildStatus/icon?job=Hyrise/hyrise-mp19/master)](https://hyrise-ci.epic-hpi.de/blue/organizations/jenkins/hyrise%2Fhyrise/activity/)
```
Comment (Bouncner) [inline README.md]: ??

File: resources/benchmark/tpcds/tables/reason.csv.json [hunk 2]
```diff
@@ -0,0 +1,29 @@
+{
```
Comment (Bouncner) [inline resources/benchmark/tpcds/tables/reason.csv.json]: Why are these JSON files part of this PR?

File: resources/test_data/sqlite_testrunner_queries.sql [hunk 3]
```diff
@@ -191,6 +191,7 @@ SELECT a, SUM(b) FROM mixed GROUP BY a;
 SELECT a, SUM(b), AVG(c) FROM mixed GROUP BY a;
 SELECT a, b, MAX(c), AVG(b) FROM mixed GROUP BY a, b;
 SELECT a AS whatever, SUM(b) FROM mixed GROUP BY whatever;
+SELECT a AS whatever, SUM(b) FROM mixed GROUP BY a;
```
Comment (Bouncner) [inline resources/test_data/sqlite_testrunner_queries.sql]: Maybe something like 'a_alias' instead of 'whatever'?

File: src/lib/sql/sql_identifier_resolver.cpp [hunk 4]
```diff
@@ -48,19 +53,24 @@ std::shared_ptr<AbstractExpression> SQLIdentifierResolver::resolve_identifier_re
   return matching_expressions[0];
 }
 
-const std::optional<SQLIdentifier> SQLIdentifierResolver::get_expression_identifier(
-    const std::shared_ptr<AbstractExpression>& expression) const {
+const std::vector<SQLIdentifier> SQLIdentifierResolver::get_expression_identifiers(
+    const std::shared_ptr<AbstractExpression> &expression) const {
```
Comment (Bouncner) [inline src/lib/sql/sql_identifier_resolver.cpp]: The position of the ampersand looks like the linter/formatter haven't been executed.

File: src/lib/operators/jit_operator/jit_types.cpp [hunk 5]
```diff
@@ -38,8 +38,7 @@ namespace opossum {
   }
 
 #define JIT_VARIANT_VECTOR_GET_VECTOR(r, d, type)                                                                 \
-  template <>                                                                                                     \
-  /* NOLINTNEXTLINE(bugprone-macro-parentheses) - clang-tidy does not like & for reference */                     \
+  template <> /* NOLINTNEXTLINE(bugprone-macro-parentheses) - clang-tidy does not like & for reference */         \
```
Comment (janetzki) [inline src/lib/operators/jit_operator/jit_types.cpp]: Is this auto-formatting problematic?

File: src/lib/sql/sql_translator.cpp [hunk 6]
```diff
@@ -228,8 +228,8 @@ std::shared_ptr<AbstractLQPNode> SQLTranslator::_translate_select_statement(cons
   if (select.limit) _translate_limit(*select.limit);
 
   /**
-   * Name, select and arrange the Columns as specified in the SELECT clause
-   */
+     * Name, select and arrange the Columns as specified in the SELECT clause
```
Comment (janetzki) [inline src/lib/sql/sql_translator.cpp]: Why does the auto-formatter do this?

Comment (Bensk1) [reply src/lib/sql/sql_translator.cpp]: Not sure, which version did you use?

File: resources/test_data/sqlite_testrunner_queries.sql [hunk 7]
```diff
@@ -191,7 +191,8 @@ SELECT * FROM id_int_int_int_100 AS r WHERE a < (SELECT SUM(min_a) FROM (SELECT
 SELECT a, SUM(b) FROM mixed GROUP BY a;
 SELECT a, SUM(b), AVG(c) FROM mixed GROUP BY a;
 SELECT a, b, MAX(c), AVG(b) FROM mixed GROUP BY a, b;
-SELECT a AS whatever, SUM(b) FROM mixed GROUP BY whatever;
+SELECT a AS a_alias, SUM(b) FROM mixed GROUP BY a_alias;
```
Comment (mrzzzrm) [inline resources/test_data/sqlite_testrunner_queries.sql]: Some more queries would be justified here, imo. What about the aliases being used in WHERE? in HAVING? is it legal to use them there?

File: src/lib/sql/sql_identifier_resolver.cpp [hunk 8]
```diff
@@ -48,19 +53,24 @@ std::shared_ptr<AbstractExpression> SQLIdentifierResolver::resolve_identifier_re
   return matching_expressions[0];
 }
 
-const std::optional<SQLIdentifier> SQLIdentifierResolver::get_expression_identifier(
+const std::vector<SQLIdentifier> SQLIdentifierResolver::get_expression_identifiers(
```
Comment (mrzzzrm) [inline src/lib/sql/sql_identifier_resolver.cpp]: I know the `const` was there before, but pls remove it. const of return-by-value makes no sense.

File: src/lib/sql/sql_identifier_resolver.hpp [hunk 9]
```diff
@@ -14,7 +14,7 @@ class AbstractExpression;
 
 struct SQLIdentifierContextEntry final {
   std::shared_ptr<AbstractExpression> expression;
-  std::optional<SQLIdentifier> identifier;
+  std::vector<SQLIdentifier> identifiers;
```
Comment (mrzzzrm) [inline src/lib/sql/sql_identifier_resolver.hpp]: Can multiple identifiers of an expression have different `table_name`s? if not, wouldn't it be cleaner to make this 
```c++
struct SQLIdentifierContextEntry final {
  std::shared_ptr<AbstractExpression> expression;
  std::optional<std::string> table_name;
  std::vector<std::string> column_names;
}
```

if it is possible that there are multiple table names for one expression, please add SQLiteTestRunner and SQLTranslatorTest - queries for it.

File: src/lib/sql/sql_identifier_resolver.cpp [hunk 10]
```diff
@@ -7,38 +7,43 @@ using namespace std::string_literals;  // NOLINT
...
-  if (entry.identifier) {
-    entry.identifier->column_name = column_name;
-  } else {
-    entry.identifier.emplace(column_name);
-  }
+  entry.identifiers.emplace_back(column_name);
+}
+
+void SQLIdentifierResolver::reset_column_names(const std::shared_ptr<opossum::AbstractExpression>& expression) {
+  auto& entry = _find_or_create_expression_entry(expression);
```
Comment (mrzzzrm) [inline src/lib/sql/sql_identifier_resolver.cpp]: so, an Entry is potentially created if it didn't exist before. The function name doesn't reflect this. Do we even want this? If not, an Assert should be added, ensuring that an Entry for `expression` existed before. 
If we want it, the function name should reflect this behaviour and queries triggering this path should be added to the test suite.

File: src/test/sql/sql_translator_test.cpp [hunk 11]
```diff
@@ -1080,6 +1095,23 @@ TEST_F(SQLTranslatorTest, FromColumnAliasingSimple) {
...
+  const auto expected_lqp =
+  AliasNode::make(expression_vector(int_float_a, int_float_b), std::vector<std::string>({"b", "a"}),
+    PredicateNode::make(equals_(int_float_a, int_float_b),
+      stored_table_node_int_float));
+  // clang-format on
+
+  EXPECT_LQP_EQ(actual_lqp_a, expected_lqp);
+  EXPECT_LQP_EQ(actual_lqp_b, expected_lqp);
+}
+
```
Comment (mrzzzrm) [inline src/test/sql/sql_translator_test.cpp]: (random line)
`SELECT a AS b, b AS a FROM t WHERE a > 5` should now yield an InputError, right? If not, add tests for it. If so, add it to `TEST_F(SQLTranslatorTest, CatchInputErrors)`

File: src/test/sql/sql_translator_test.cpp [hunk 12]
```diff
@@ -492,6 +492,21 @@ TEST_F(SQLTranslatorTest, AggregateWithGroupBy) {
   EXPECT_LQP_EQ(actual_lqp, expected_lqp);
 }
 
+TEST_F(SQLTranslatorTest, AliasWithGroupBy) {
+  const auto actual_lqp = compile_query("SELECT a AS x FROM int_float GROUP BY a");
```
Comment (mrks) [inline src/test/sql/sql_translator_test.cpp]: Also test `GROUP BY x`?

File: src/lib/sql/sql_identifier_resolver.hpp (hunk 9 above)
Comment (mrks) [inline src/lib/sql/sql_identifier_resolver.hpp]: Adding on @mrzzzrm comment:

What is the result of `SELECT a AS x FROM t GROUP BY t.x`?

File: src/lib/sql/sql_identifier_resolver.hpp [hunk 13]
```diff
@@ -25,9 +25,10 @@ class SQLIdentifierResolver final {
  public:
   /**
    * @{
-   * Set/Update the column/table name of an expression
+   * Set/Update/Delete the column/table names of an expression
    */
-  void set_column_name(const std::shared_ptr<AbstractExpression>& expression, const std::string& column_name);
+  void add_column_name(const std::shared_ptr<AbstractExpression>& expression, const std::string& column_name);
```
Comment (mrks) [inline src/lib/sql/sql_identifier_resolver.hpp]: I'd add a comment explaining why an expression might have multiple column names

File: src/lib/sql/sql_identifier_resolver.cpp (hunk 8 above)
Comment (mrks) [inline src/lib/sql/sql_identifier_resolver.cpp]: Wouldn't it be cheaper to return a reference here?

File: src/lib/sql/sql_translator.cpp [hunk 14]
```diff
@@ -522,8 +525,9 @@ SQLTranslator::TableSourceState SQLTranslator::_translate_table_origin(const hsq
                 "Must specify a name for exactly each column");
 
     for (auto column_id = ColumnID{0}; column_id < hsql_table_ref.alias->columns->size(); ++column_id) {
-      sql_identifier_resolver->set_column_name(column_expressions[column_id],
-                                               (*hsql_table_ref.alias->columns)[column_id]);
+      const auto& expression = column_expressions[column_id];
+      sql_identifier_resolver->reset_column_names(expression);
```
Comment (mrks) [inline src/lib/sql/sql_translator.cpp]: Why does it have to be reset?

File: resources/test_data/sqlite_testrunner_queries.sql (hunk 7 above)
Comment (mrks) [reply resources/test_data/sqlite_testrunner_queries.sql]: I'd also add tests that make sure that the aliases properly survive joins etc.

File: src/lib/sql/sql_translator.cpp (hunk 6 above)
Comment (janetzki) [reply src/lib/sql/sql_translator.cpp]: 7.0.1-8

File: src/lib/sql/sql_identifier_resolver.hpp (hunk 9 above)
Comment (janetzki) [reply src/lib/sql/sql_identifier_resolver.hpp]: Having multiple identifiers for the same expression with different `table_name`s could work for a query like:
```sql
SELECT R.a, S.a FROM mixed AS R, mixed AS S
```
However, the `SQLTranslator` does not support this behavior since it creates two `StoredTableNode`s in `_translate_stored_table()`. Therefore, `R.a` and `S.a` are identifiers for different expressions. Do you think that changing the SQLTranslator makes sense?
I also added tests for the mentioned query.

Comment (mrks) [reply src/lib/sql/sql_identifier_resolver.hpp]: On a general note: As long as we don't break anything, we don't necessarily have to cover edge cases that we don't see in the TPC-DS

File: resources/test_data/sqlite_testrunner_queries.sql (hunk 7 above)
Comment (janetzki) [reply resources/test_data/sqlite_testrunner_queries.sql]: Do we want to support queries like `SELECT a AS x FROM mixed WHERE x > 10`? The `SQLTranslator` does not support this at the moment.

Comment (mrks) [reply resources/test_data/sqlite_testrunner_queries.sql]: I believe that this is inconsistent across other DBMSs. My thoughts are:

1. If we need it for TPC-DS and it is reasonably doable, do it.
2. If it's not part of the TPC-DS syntax but can be tacked on without significant extra effort, do it.
3. If adding it adds significant effort, don't.
3a. If possible, add new assertions so that we, should it ever come around, get a more helpful message (maybe even with a comment outlining what would have to be done)

File: src/lib/sql/sql_identifier_resolver.hpp (hunk 9 above)
Comment (mrzzzrm) [reply src/lib/sql/sql_identifier_resolver.hpp]: > However, the SQLTranslator does not support this behavior since it creates two StoredTableNodes in _translate_stored_table().

That is fundamentally intended behavior. R.a and S.a are different columns/expressions in this case. They just happen to come from the same table. Think of `SELECT R.a, S.a FROM mixed AS R, mixed AS S WHERE S.a > 5` - it is obvious that R.a and S.a now contain different data - so they cannot possibly be the same expression/column

File: resources/test_data/sqlite_testrunner_queries.sql (hunk 7 above)
Comment (mrzzzrm) [reply resources/test_data/sqlite_testrunner_queries.sql]: I imagine this requires effort and agree with Markus' other points

File: src/lib/sql/sql_identifier_resolver.hpp (hunk 9 above)
Comment (janetzki) [reply src/lib/sql/sql_identifier_resolver.hpp]: Thanks for the clarification. Handling these columns identically would not help us implementing the TPC-DS. Thus, I won't change the `SQLTranslator`.

**Comment by mrzzzrm** [comment]
@janetzki once you feel you have addressed the review, please ping me and I'll make another pass.

File: src/lib/sql/sql_identifier_resolver.hpp (hunk 9 above)
Comment (janetzki) [reply src/lib/sql/sql_identifier_resolver.hpp]: This crashes, as well as SQLite. I added a test for this.

File: src/lib/sql/sql_translator.cpp (hunk 14 above)
Comment (janetzki) [reply src/lib/sql/sql_translator.cpp]: This is especially for queries like `SELECT * FROM int_float AS t (b, a) WHERE b = t.a`, in which the original column names should not be accessible anymore.

File: resources/test_data/sqlite_testrunner_queries.sql (hunk 7 above)
Comment (janetzki) [reply resources/test_data/sqlite_testrunner_queries.sql]: I was able to add alias support in the WHERE clause (https://github.com/hyrise-mp/hyrise/commit/6d9c7e8b7a642ba97a2cd845914ea773911f2bf5). However, it has some consequences:

Comment (janetzki) [reply resources/test_data/sqlite_testrunner_queries.sql]: >     2. If it's not part of the TPC-DS syntax but can be tacked on without significant extra effort, do it.
Yes, it's not required for the TPC-DS.

Comment (mrzzzrm) [reply resources/test_data/sqlite_testrunner_queries.sql]: I don't see a clear win here and it breaks even more with this comment, though I wouldn't see this as important unless the referenced blog post would actually reference the standard.

```
  // SQL Orders of Operations: http://www.bennadel.com/blog/70-sql-query-order-of-operations.htm
  // 1. FROM clause (incl. JOINs and sub-SELECTs that are part of this)
  // 2. WHERE clause
  // 3. GROUP BY clause
  // 4. HAVING clause
  // 5. SELECT clause (incl. DISTINCT)
  // 6. UNION clause
  // 7. ORDER BY clause
  // 8. LIMIT clause
```

We gain one thing and loose another - I'm fine with it. 

> This does not seem to be a problem from my perspective. I only wondered because there is an explicit test for this

Don't worry, the ParamaterIDs are arbitrary, they just have to be unique. `parameter_ids_of_value_placeholders` is what gives them order. Explicit values were tested to keep things simple.

Unless @mrks says something else, I'd say we can keep this. Please revise the naming of `_translate_select_list_groupby_having` as its behavior has  changed. Adjust that comment I mentioned above (probably removing the URL)

Comment (mrks) [reply resources/test_data/sqlite_testrunner_queries.sql]: Sounds good.

> [...] not by Hyrise anymore because a becomes ambiguous.

So be it. Whoever writes those queries deserves to be punished. ;)

File: src/test/sql/sql_translator_test.cpp (hunk 11 above)
Comment (janetzki) [reply src/test/sql/sql_translator_test.cpp]: Yes, I added a test for the case that this query yields an 'InvalidInputException'.

**Comment by janetzki** [comment]
@mrzzzrm I addressed all your comments. Please have another look.

File: src/lib/sql/sql_identifier_resolver.cpp [hunk 15]
```diff
@@ -48,19 +53,25 @@ std::shared_ptr<AbstractExpression> SQLIdentifierResolver::resolve_identifier_re
...
 
-const std::optional<SQLIdentifier> SQLIdentifierResolver::get_expression_identifier(
+const std::vector<SQLIdentifier>& SQLIdentifierResolver::get_expression_identifiers(
     const std::shared_ptr<AbstractExpression>& expression) const {
   auto entry_iter = std::find_if(_entries.begin(), _entries.end(),
                                  [&](const auto& entry) { return *entry.expression == *expression; });
-  if (entry_iter == _entries.end()) return std::nullopt;
-  return entry_iter->identifier;
+  const auto empty_vector = std::make_shared<std::vector<SQLIdentifier>>();
+  if (entry_iter == _entries.end()) return *empty_vector;
```
Comment (mrzzzrm) [inline src/lib/sql/sql_identifier_resolver.cpp]: This returns a reference to a local variable, which is subsequently destroyed. Did none of our sanitizers catch this?

Comment (mrzzzrm) [reply src/lib/sql/sql_identifier_resolver.cpp]: I think I'd be fine with return-by-value.

File: src/lib/sql/sql_identifier_resolver.hpp (hunk 9 above)
Comment (mrzzzrm) [reply src/lib/sql/sql_identifier_resolver.hpp]: why is this resolved? I believe the answer to my original question ("Can multiple identifiers of an expression have different table_names?") was "No", so wouldn't the layout of `SQLIdentifierContextEntry` that I proposed be cleaner?

File: src/lib/sql/sql_identifier_resolver.hpp [hunk 16]
```diff
@@ -25,9 +25,11 @@ class SQLIdentifierResolver final {
  public:
   /**
    * @{
-   * Set/Update the column/table name of an expression
+   * Set/Update/Delete the column/table names of an expression. There can be multiple column names because a new alias
```
Comment (mrzzzrm) [inline src/lib/sql/sql_identifier_resolver.hpp]: "There can be multiple column names *referring to a single expression*"?

File: src/lib/sql/sql_translator.cpp (hunk 14 above)
Comment (mrzzzrm) [reply src/lib/sql/sql_translator.cpp]: Please add this as a comment, I think that example helps.

File: src/lib/sql/sql_translator.cpp [hunk 17]
```diff
@@ -739,7 +748,31 @@ SQLTranslator::TableSourceState SQLTranslator::_translate_cross_product(const st
   return result_table_source_state;
 }
 
-void SQLTranslator::_translate_select_list_groupby_having(const hsql::SelectStatement& select) {
+std::vector<std::shared_ptr<AbstractExpression>> SQLTranslator::_translate_select_list(
+    const hsql::SelectStatement& select) {
```
Comment (mrzzzrm) [inline src/lib/sql/sql_translator.cpp]: Could take whatever the type of `select.selectList` is as a parameter?

File: src/test/sql/sql_translator_test.cpp [hunk 18]
```diff
@@ -492,6 +510,81 @@ TEST_F(SQLTranslatorTest, AggregateWithGroupBy) {
   EXPECT_LQP_EQ(actual_lqp, expected_lqp);
 }
 
+TEST_F(SQLTranslatorTest, AliasWithGroupBy) {
```
Comment (mrzzzrm) [inline src/test/sql/sql_translator_test.cpp]: I'd call these tests "SelectListAliasUsedIn...."

File: src/test/sql/sql_translator_test.cpp [hunk 19]
```diff
@@ -1106,6 +1216,20 @@ TEST_F(SQLTranslatorTest, FromColumnAliasingTablesSwitchNames) {
   EXPECT_LQP_EQ(actual_lqp_b, expected_lqp);
 }
 
+TEST_F(SQLTranslatorTest, SameColumnForDifferentTableNames) {
+  const auto actual_lqp = compile_query("SELECT R.a, S.a FROM int_float AS R, int_float AS S");
+
+  // clang-format off
+  const auto expected_lqp =
+  ProjectionNode::make(expression_vector(int_float_a, int_float_a),
+      JoinNode::make(JoinMode::Cross,
```
Comment (mrzzzrm) [inline src/test/sql/sql_translator_test.cpp]: indentation is off

**Comment by mrzzzrm** [comment]
@mrks Once these points are addressed, I am fairly happy with this. Won't be available until Tuesday, feel free to merge without my approval.

File: src/lib/sql/sql_identifier_resolver.cpp (hunk 15 above)
Comment (janetzki) [reply src/lib/sql/sql_identifier_resolver.cpp]: Neither `lint.sh` not `format.sh` did something about this. Shouldn't the object be kept until there is no more reference to it because it comes from a shared pointer?

File: src/lib/sql/sql_translator.cpp (hunk 14 above)
Comment (janetzki) [reply src/lib/sql/sql_translator.cpp]: I added a shorter version of this comment because there already is an explanation a few lines earlier.

File: src/lib/sql/sql_identifier_resolver.hpp (hunk 16 above)
Comment (janetzki) [reply src/lib/sql/sql_identifier_resolver.hpp]: Good catch. It should make more sense now.

File: src/lib/sql/sql_identifier_resolver.cpp (hunk 15 above)
Comment (mrks) [reply src/lib/sql/sql_identifier_resolver.cpp]: `*empty_vector` does not increase the counter, so there is exactly one `shared_ptr` pointing to `empty_vector`. When that `shared_ptr` goes out of scope, `*empty_vector` is broken.

My approach for things like this is to have a `static` empty object and return a const reference to that one.

File: src/lib/sql/sql_identifier_resolver.hpp (hunk 16 above)
Comment (mrks) [reply src/lib/sql/sql_identifier_resolver.hpp]: I'd expect it to be a suggestion

File: src/lib/sql/sql_identifier_resolver.cpp (hunk 15 above)
Comment (mrzzzrm) [reply src/lib/sql/sql_identifier_resolver.cpp]: > Neither lint.sh not format.sh did something about this

I was referring to valgrind, ASAN and UBSAN, but since this PR doesn't have FullCI enabled yet, those haven't actually run yet.

@mrks' solution is fine (though I always wondered whether there might be some performance hit when using it in multi-threaded scenarios)

Comment (mrks) [reply src/lib/sql/sql_identifier_resolver.cpp]: You don't get any software level parallelization guarantees. If you want to access it from multiple threads, you will have to synchronize the access yourself. As the object is not modified, there should also be no cache line synchronization necessary. A `const` wouldn't hurt though.

Comment (mrzzzrm) [reply src/lib/sql/sql_identifier_resolver.cpp]: Oh, but you do, right? 

https://stackoverflow.com/a/17713799

"...static storage duration for a function local means that only one instance of that local exists in the program" 

"The thread safety bits make it get a bit more complicated..."

But looking at the assembly in the SO answer, the primary guard variable is not synchronized as far as I can tell, so the overhead *should* be minimal

File: src/lib/sql/sql_identifier_resolver.hpp (hunk 9 above)
Comment (mrks) [reply src/lib/sql/sql_identifier_resolver.hpp]: @mrzzzrm I checked the other open points, which are addressed. With this one, I am not sure enough to discard your change request. Could you have a look at https://github.com/hyrise/hyrise/pull/1693/commits/67fb153dd8cd8cffdd105f062c2988eb0dcba1df

File: src/lib/sql/sql_identifier_resolver.cpp (hunk 15 above)
Comment (janetzki) [reply src/lib/sql/sql_identifier_resolver.cpp]: I chose to do it return-by-value because the `SQLIdentifier`s are not part of the resolver anymore, but they are created ad hoc.

File: src/lib/sql/sql_identifier_resolver.hpp (hunk 9 above)
Comment (janetzki) [reply src/lib/sql/sql_identifier_resolver.hpp]: Yes, this commit addresses your suggestion @mrzzzrm. (Sorry for resolving the discussion too soon, though.) As a consequence, we can now run queries, such as `SELECT a AS x FROM int_float GROUP BY int_float.x`, which does not look right in my eyes but is unambiguous.

**Comment by janetzki** [comment]
Thanks a lot for your critical reviews @mrzzzrm! I addressed all your comments.

File: src/lib/sql/sql_identifier_resolver.hpp (hunk 9 above)
Comment (mrzzzrm) [reply src/lib/sql/sql_identifier_resolver.hpp]: Ok, I believe that's acceptable @mrks ?

Comment (mrks) [reply src/lib/sql/sql_identifier_resolver.hpp]: Doesn’t feel completely right, but I don’t expect it to hurt anywhere.

**Comment by mrks** [comment]
Could someone merge this once it’s green?
